pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1116,6 +1192,70 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func setCgroupCpuset(config *types.DomainConfig, status *types.DomainStatus) error {
+	cgroupName := filepath.Join(containerd.GetServicesNamespace(), config.GetTaskName())
+	cgroupPath := cgroups.StaticPath(cgroupName)
+	controller, err := cgroups.Load(cgroups.V1, cgroupPath)
+	if err != nil {
+		// It's still not an error, since the path may still not exist
+		log.Warnf("Failed to find cgroups directory for %s", config.DisplayName)
+		return nil
+	}
+	err = controller.Update(&specs.LinuxResources{CPU: &specs.LinuxCPU{Cpus: status.VmConfig.CPUs}})
+	if err != nil {
+		log.Warnf("Failed to update CPU set for %s", config.DisplayName)
+		return err
+	}
+	log.Functionf("Adjust the cgroups cpuset of %s to %s", config.DisplayName, status.VmConfig.CPUs)
+	return nil
+}
+
+func constructNonPinnedCpumaskString(ctx *domainContext) string {
```

Comment:
As I understand, the function will return at least reserved CPU for EVE-OS, if other CPUs are pinned. Ia it correct? Probably it make sense to mention it somewhere in comment, or check again if we have empty string as result, which may broke the logic.

Reply:
Yes, that's right. We have the reserved CPUs always free so that they will be in the resulting string anyway.
Added a comment.